### ISSUE TITLE
Typescript autocomplete broken with latest typescript version (4.4.2)

### ISSUE THREAD

**Issue by dhushyanth-s** [modulz/stitches#759]
# Bug report
The autocomplete is broken for the `styled` function with typescript versions ^4.4.2 which is the latest stable version.

## To Reproduce
1. Create a new react app with typescript
```zsh
npx create-react-app --template typescript
```

2. Add stitches to dependencies
```zsh
npm i @stitches/react
```

3. Import stitches with
```ts
import { styled } from '@stitches/react'
```

4. Try invoking autocomplete for css attributes.

## Expected behavior

All css attributes should be shown in autocomplete menu

## Screenshots

with TS compiler version 4.4.2
<img width="524" alt="image" src="https://user-images.githubusercontent.com/54908910/132107086-355beb80-d430-42f5-a90c-df885562d02a.png">

with TS version 4.3.5 (from stitches repo `package.json` file)
<img width="517" alt="image" src="https://user-images.githubusercontent.com/54908910/132107108-5994088d-92be-4070-a1d1-c85e051efce7.png">

## System information

- OS: macOS Big Sur 11.4 on M1 Macbook Pro
- Version of Stitches: 1.0.0
- Version of Node.js: v16.5.0

## Additional context

 - VSCode version: 1.60.0

**Comment by peduarte** [comment]
Hi! 

I've just followed your steps and was unable to reproduce:
![CleanShot 2021-09-06 at 13 18 39@2x](https://user-images.githubusercontent.com/372831/132209513-83940446-dafc-43f4-a0b4-d980f0260176.jpg)



The only difference is that I'm not on an M1 Mac, on Intel.



Thanks

**Comment by dhushyanth-s** [comment]
As you can see on the bottom right of the VSCode window, VSCode is using typescript version 4.3.5 which is working as expected please test with latest stable version 4.4.2 or later. 

You can change IDE typescript versions by just clicking on the version number in the status bar of the window.

**Comment by peduarte** [comment]
@dhushyanth-s ah, yep, you're right. Thank you for the heads up!

I am able to replicate this issue now.

**Comment by cpakken** [comment]
I have the same issue as well

**Comment by cpakken** [comment]
Typescript 4.4 added control flow analysis, maybe this change is causing the issue?
https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#cfa-aliased-conditions.

Also even in Typescript 4.4.2, the autocomplete works when defining utils inside createStitches: 
![image](https://user-images.githubusercontent.com/61718569/132591949-688bcaf8-b1a7-491f-a58e-514d6f5e9438.png)

Hope this is useful for the bugfix

**Comment by peduarte** [comment]
Hi! This has been released to Stitches v1.1.0

```
yarn add @stitches/react@latest
# or
npm install @stitches/react@latest
```

**Comment by dhushyanth-s** [comment]
Wow that was quick!
